mlir/lib/Dialect/LLVMIR/Transforms/DataLayoutFromTarget.cpp

```diff
@@ -0,0 +1,61 @@
+//===- DataLayoutFromTarget.cpp - extract data layout from TargetMachine --===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/DLTI/DLTI.h"
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Pass/Pass.h"
+
+namespace mlir {
+namespace LLVM {
+#define GEN_PASS_DEF_LLVMDATALAYOUTFROMTARGET
+#include "mlir/Dialect/LLVMIR/Transforms/Passes.h.inc"
+} // namespace LLVM
+} // namespace mlir
+
+using namespace mlir;
+
+struct DataLayoutFromTargetPass
+    : public LLVM::impl::LLVMDataLayoutFromTargetBase<
+          DataLayoutFromTargetPass> {
+  void runOnOperation() override {
+    ModuleOp mod = getOperation();
+
+    bool passFailed = false;
+
+    mod->walk([&](ModuleOp mod) {
+      auto targetAttr =
+          mod->getAttrOfType<LLVM::TargetAttrInterface>("llvm.target");
```

Comment:
We shouldn't have a magic strings, can you define "llvm.target" as a `let discardableAttrs = (ins ...` on the dialect instead?